src/meta-srv/src/metasrv.rs

```diff
@@ -357,13 +358,27 @@ impl MetaSrv {
             self.leader_cached_kv_backend
                 .load()
                 .await
-                .context(error::KvBackendSnafu)?;
+                .context(KvBackendSnafu)?;
             self.procedure_manager
                 .start()
                 .await
                 .context(StartProcedureManagerSnafu)?;
         }
 
+        if self
+            .kv_backend
+            .exists(MAINTENANCE_KEY.as_bytes())
+            .await
+            .context(KvBackendSnafu)?
+        {
+            let req = PutRequest {
+                key: Vec::from(MAINTENANCE_KEY),
+                value: vec![],
+                prev_kv: false,
+            };
+            self.in_memory.put(req).await.context(KvBackendSnafu)?;
+        }
+
```

Comment:
If we rely on the `in_memory` component,  We need to move these code to [here](https://github.com/GreptimeTeam/greptimedb/blob/74862f8c3fc083545d9c95f91e36b2f3190dfe21/src/meta-srv/src/metasrv.rs#L212-L229); Whenever a new leader comes, it should re-load this key value from the remote.

Otherwise, we should use the `LeaderCachedKvBackend `; it doesn't require re-loading the key value. However, remember that only the leader of MetaSrv can write key values for it.

Reply:
We can directly use KvBackend(Etcd), it will work well.